src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -153,7 +236,10 @@ winrt::com_ptr<GlobalAppSettings> GlobalAppSettings::FromJson(const Json::Value&
 
 void GlobalAppSettings::LayerJson(const Json::Value& json)
 {
-    JsonUtils::GetValueForKey(json, DefaultProfileKey, _unparsedDefaultProfile);
+    // _validDefaultProfile keeps track of whether we've verified that DefaultProfile points to something
+    // CascadiaSettings::_ResolveDefaultProfile performs a validation and updates DefaultProfile() with the
+    // resolved value, then making it valid.
+    _validDefaultProfile = !JsonUtils::GetValueForKey(json, DefaultProfileKey, _UnparsedDefaultProfile);

```

Comment:
Right. Changed this to:
```c++
    _validDefaultProfile = false;
    JsonUtils::GetValueForKey(json, DefaultProfileKey, _UnparsedDefaultProfile);
```
`_validDefaultProfile` _needs_ to be reset to false when we import a new value via the json. This is because `_defaultProfile` is not populated yet (only `_UnparsedDefaultProfile`). In `CascadiaSettings`, we resolve/validate the provided value in `ResolveDefaultProfile` and `ValidateDefaultProfileExists`. So at this point, `_defaultProfile` can be extracted and used properly.

For the purposes of the Settings UI, we should only ever interact with the `UnparsedDefaultProfile` to give the user the flexibility of using a guid or name as an identifier. Eventually, we'll make it so that SUI just gives you a list of profiles to choose from, and this extra level of complication can maybe go away with a refactor?